src/d/actor/d_a_npc_bs1.cpp

```diff
@@ -137,161 +567,651 @@ void daNpc_Bs1_c::shop_talk() {
+                            (s16)0x800);
+    }
+    mJntCtrl.lookAtTarget(&current.angle.y, &destPos, other, desiredYRot, m724, true);
 }
 
 /* 000037B8-00003830       .text wait01__11daNpc_Bs1_cFv */
-void daNpc_Bs1_c::wait01() {
+BOOL daNpc_Bs1_c::wait01() {
     /* Nonmatching */
+    if (m731 != 0) {
+        m831 = m830;
+        m830 = 2;
+    } else {
+        if (m730 != 0) {
+            m82A = 2;
+            // cmplwi but is cmpwi
+        } else if (m82A == 0x02) {
+            m82A = 0;
+        }
+    }
+    // Return is also not matching
+    return mpMorf->mCurMorf < 1.0f;
 }
 
 /* 00003830-00003970       .text talk01__11daNpc_Bs1_cFv */
-void daNpc_Bs1_c::talk01() {
-    /* Nonmatching */
+BOOL daNpc_Bs1_c::talk01() {
+    u16 value = talk();
+    if (value == 0x12) {
+        daPy_py_c* player = daPy_getPlayerActorClass();
+        m830 = m831;
+        dComIfGp_event_reset();
+        m731 = 0;
+        setAnm(J3DFrameCtrl::LOOP_ONCE_e);
+        player->offPlayerNoDraw();
+        mShopCamAction.Reset();
+    } else {
+        if (shopMsgCheck(m738) && value == 8) {
+            if (mShopItems.getSelectItemBuyMsg() == m738) {
+                dComIfGp_setm4932(0x17);
+                dComIfGp_setm4931(0x27);
+            }
+        } else if (value == 8 && checkBeastItemSellMsg(m738)) {
+            dComIfGp_setm4932(0x17);
+            dComIfGp_setm4931(0x27);
+        }
+    }
+    return mpMorf->mCurMorf < 1.0f;
 }
 
 /* 00003970-00003A70       .text wait_action__11daNpc_Bs1_cFPv */
-void daNpc_Bs1_c::wait_action(void*) {
+BOOL daNpc_Bs1_c::wait_action(void*) {
     /* Nonmatching */
+    if (m834 == 0) {
+        m830 = 1;
+        m731 = 0;
+        m834 += 1;
+    } else if (m834 != -1) {
+        s16 val = current.angle.y + getHead_y() + getBackbone_y();
+        m730 = chkAttention(current.pos, val);
+        BOOL setAttn;
+        switch (m830) {
+        case 1:
+            setAttn = wait01();
+            break;
+        case 2:
+            setAttn = talk01();
+            break;
+        default:
+            setAttn = FALSE;
+        }
+        lookBack();
+        // If I make setAttn a primitive bool, the nonmatching goes to other functions
+        setAttention(setAttn);
+    }
+    return TRUE;
 }
 
 /* 00003A70-00003CB4       .text getdemo_action__11daNpc_Bs1_cFPv */
-void daNpc_Bs1_c::getdemo_action(void*) {
+BOOL daNpc_Bs1_c::getdemo_action(void*) {
     /* Nonmatching */
+    u32 staffIdx = dComIfGp_evmng_getMyStaffId(&a_name[m834]);
+    u32 actIdx = dComIfGp_evmng_getMyActIdx(staffIdx, &a_cut_name, 2, 0, 0);
+    if (m834 == 0) {
+        daPy_getPlayerActorClass()->offPlayerNoDraw();
+        m830 = m831;
+        mShopCamAction.Reset();
+        u16 itemNo = mShopItems.getSelectItemNo();
+        uint itemPtr =
+            fopAcM_createItemForPresentDemo(&current.pos, (u8)itemNo, 0, -1, current.roomNo);
+        if (itemPtr != -1) {
```

Comment:
```suggestion
        if (itemPID != fpcM_ERROR_PROCESS_ID_e) {
```
fopAcM_createItemForPresentDemo returns the process ID of the created item, not a pointer to the item itself (as it isn't created immediately when it's requested)